test/residual/arrayExpression.js

```diff
@@ -0,0 +1 @@
+[1, "2", [3], ...[4, 5]]
```

Comment:
I'd like to see a better testing story. Having dedicated unit tests is nice, but... Shouldn't you be able to just run all of the test262 tests through this new partial evaluator?
We just didn't do that for the serializer tests, as the test262 just didn't exercise the serializer much, and we didn't have a good oracle.